tests/Checker/Check_Result_Tests.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Tests for the Check_Result class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+class Check_Result_Tests extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+
+		$check_context = new Check_Context( 'test-plugin/test-plugin.php' );
+
+		$this->check_result = new Check_Result( $check_context );
+	}
+
+	public function test_plugin() {
+		$this->assertInstanceOf( Check_Context::class, $this->check_result->plugin() );
+	}
+
+	public function test_add_message() {
+		$this->check_result->add_message(
+			false,
+			'Warning message',
+			array(
+				'code'   => 'test_warning',
+				'file'   => 'test-plugin/test-plugin.php',
+				'line'   => 12,
+				'column' => 40,
+			)
+		);
+
+		$warnings = $this->check_result->get_warnings();
+
+		// Tests the filename used as the main key for the message associated with that file.
+		$this->assertArrayHasKey( 'test-plugin.php', $warnings );
+
+		// Tests the line number is used as the first key for the filename array.
+		$this->assertArrayHasKey( 12, $warnings['test-plugin.php'] );
+
+		// Tests the column is used as the first key for the line number array.
+		$this->assertArrayHasKey( 40, $warnings['test-plugin.php'][12] );
+
+		// Tests the column array contains the message details.
+		$expected = array(
+			'message' => 'Warning message',
+			'code'    => 'test_warning',
+		);
+
+		$this->assertEquals( $expected, $warnings['test-plugin.php'][12][40][0] );
+	}
```

Comment:
Isn't this test pretty much the same as the one below? In that case let's remove it.

Reply:
Thanks @felixarntz this was very similar so I have removed this test